scripts/upload.py

```diff
@@ -27,8 +29,15 @@ def get_unique_name(filename: str, unique_id: str) -> str:
 
 def upload(globpath: str, container: str, queue: str, sas_token_env: str, storage_account_uri: str):
     try:
-        sas_token = os.getenv(sas_token_env)
-        if sas_token is None:
+        credential = None
+        try:
+            dac = DefaultAzureCredential()
+            credential = ClientAssertionCredential(TENANT_ID, CLIENT_ID, lambda: dac.get_token("api://AzureADTokenExchange/.default").token)
+            credential.get_token("https://storage.azure.com/.default")
```

Comment:
Adding this line should ensure that if we are unable to get our MI cred, we will fall back to using the sas token.